packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -2,156 +2,107 @@ import React, { FC, useState } from 'react';
   const t = useTranslation(['common', 'inventory']);
-
   const { draftLines, update, addLine, isLoading, save, nextItem } =
     useStocktakeLineEdit(currentItem);
 
-  return (
-    <Modal
-      title={
-        mode === ModalMode.Create
-          ? t('heading.add-item')
-          : t('heading.edit-item')
-      }
-      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
-      nextButton={
-        <DialogButton
-          variant="next"
-          onClick={async () => {
-            try {
-              await save(draftLines);
-              if (nextItem) setCurrentItem(nextItem);
-              else onClose();
+  const onNext = async () => {
+    await save(draftLines);
+    if (nextItem) setCurrentItem(nextItem);
+    else onClose();
+    // Returning true here triggers the slide animation
+    return true;
+  };
 
-              // Returning true here triggers the slide animation
-              return true;
-            } catch (e) {
-              //
-            }
-          }}
-          disabled={mode !== ModalMode.Update}
-        />
-      }
-      okButton={
-        <DialogButton
-          variant="ok"
-          onClick={async () => {
-            try {
-              await save(draftLines);
-              onClose();
-            } catch (e) {
-              error(t('error.cant-save'))();
-            }
-          }}
-        />
-      }
-      height={600}
-      width={1024}
+  const onOk = async () => {
+    try {
+      await save(draftLines);
+      onClose();
+    } catch (e) {
+      error(t('error.cant-save'))();
+    }
+  };
+
+  return (
+    <StocktakeLineEditModal
+      onNext={onNext}
+      onOk={onOk}
+      onCancel={onClose}
+      mode={mode}
+      isOpen={isOpen}
     >
-      {!isLoading ? (
-        <>
-          <StocktakeLineEditForm
-            item={currentItem}
-            onChangeItem={setCurrentItem}
-            mode={mode}
-          />
-          <Divider margin={5} />
-          {currentItem ? (
-            <TabContext value={currentTab}>
-              <Box flex={1} display="flex" justifyContent="space-between">
-                <Box flex={1} />
-                <Box flex={1}>
-                  <TabList
-                    value={currentTab}
-                    centered
-                    onChange={(_, v) => setCurrentTab(v)}
-                  >
-                    <Tab value={Tabs.Batch} label={Tabs.Batch} />
-                    <Tab value={Tabs.Pricing} label={Tabs.Pricing} />
-                  </TabList>
-                </Box>
-                <Box flex={1} justifyContent="flex-end" display="flex">
-                  <ButtonWithIcon
-                    color="primary"
-                    variant="outlined"
-                    onClick={addLine}
-                    label={t('label.add-batch', { ns: 'inventory' })}
-                    Icon={<PlusCircleIcon />}
-                  />
-                </Box>
-              </Box>
+      {(() => {
```

Comment:
I'm trying this out... is it confusing? 😁 

The issue being you can't write conditional JS inside the JSX markup which needs to be nested in the modal

Reply:
works for me!
Better than the nested ternary operators that were here before 🤷